src/core/ext/lb_policy/grpclb/grpclb.c

```diff
@@ -279,58 +285,84 @@ struct rr_connectivity_data {
   glb_lb_policy *glb_policy;
 };
 
+static bool process_serverlist(const grpc_grpclb_server *server,
+                               struct sockaddr_storage *sa, size_t *sa_len) {
+  if (server->port >> 16 != 0) {
+    gpr_log(GPR_ERROR, "Invalid port '%d'.", server->port);
+    return false;
+  }
+  const uint16_t netorder_port = htons((uint16_t)server->port);
+  /* the addresses are given in binary format (a in(6)_addr struct) in
+   * server->ip_address.bytes. */
+  const grpc_grpclb_ip_address *ip = &server->ip_address;
+  *sa_len = 0;
+  if (ip->size == 4) {
+    struct sockaddr_in *addr4 = (struct sockaddr_in *)sa;
+    *sa_len = sizeof(struct sockaddr_in);
+    memset(addr4, 0, *sa_len);
+    addr4->sin_family = AF_INET;
+    memcpy(&addr4->sin_addr, ip->bytes, ip->size);
+    addr4->sin_port = netorder_port;
+  } else if (ip->size == 6) {
```

Comment:
Yes :) The whole ipv*4* vs *6* and the fact that 6 is 16 sans the 1 conflated to create this blunder. I take solace in knowing that at least I got the message below right